src/src/CustomBuild/define_plugin_sets.h

```diff
@@ -990,6 +990,8 @@ To create/register a plugin, you have to :
     #define USES_P063   // TTP229_KeyPad
     #define USES_P073   // 7DG
     #define USES_P079   // Wemos Motoshield
+
+    #define USES_P250   // RTD Sensors, e.g. MAX38165
```

Comment:
Adding a plugin to the STABLE set usually results in many builds failing (the 1M units have less space for a sketch), so you'd better add it to a TESTING set, I suggest the D set, as that should be the one with the most available space.

Reply:
I will update the pull this afternoon with your final ID proposition and put the respective #define in the testing section for set D. If until then no other ID is given i will use the initially proposed 115 - sounds nice to me. In the unlikely case i will not find the right area in the header, i will get back to you.

Putting the plugin to stable was the fastest way to a test  build on my set up. I am of course fine with an adder to test build first - not expecting my code is a "perfect run" for all builds.

Reply:
In the MAX set definition there is already an entry for P115, (just search for `USES_P115`) you might want to add your description there as well, and also add a commented-out `// #define USES_P115` (with the same description) in file `src/Custom-sample.h`.

Are you sure it can't be added to P039? As it seems an extension with different hardware, and the rest of the plugin shouldn't have to change that much. But I haven't done a proper compare yet, so maybe it is too different after all. And the increased size of that plugin may break the smaller builds (again)...

Reply:
Well this one tipped the 'normal' builds over the build size limit, so I guess merging it with P039 may do the same.
On the other hand, there is quite a lot of code similarity, so I guess some code deduplication does make sense.
But maybe the shared code is also present in a number of other SPI sensor plugins, so then a single "plugin SPI helper" would be more useful.

Reply:
Yes, offer standard access functions to SPI (e.g. init SPI, read8/16BitRegisters(), write8/16BitRegisters(), CS Handling,...) in a global "SPI high level library" could make sense, as those functions will be required by all SPI plugins or functions. When starting the LM7x portion i realized that my implementation of basic read/write functions lack full re-use potential as i incorporated the CS handling to those. With that in mind i would carve out the CS handling now to a separate function to allow more re-use.

E.g. LM7x requries 64 bit consecutive access to SPI registers - limiting the consecutive bit length to 8/16 Bit between CS operation in the read/write operations does not support optimized ROM consumption...

I will analyze the merging option more thoroughly, frankly i see it possible as P039 has been the start point for my implementation. And as already mentioned i can follow the idea of grouping identical topics to plugins.

I propose to keep this PR in the current state and investigate on the merging in a separate one. 

@tonhuisman  - i saw there was a former module P115 present in an experimental state  - has this been transferred to P088 ?

--> file define plugin_sets.h, line 1233:
--> it is still referenced in _C002.ino, line166ff:

Reply:
You are referring to the EXPERIMENTAL set, in fact the experimental plugins that (mostly) are available in the [ESPEasyPlayground](https://github.com/letscontrolit/ESPEasyPluginPlayground) repository. These plugins are mostly in a pre-acceptance state, though many are quite mature and well-tested, but where never brought over to the main (mega) repository. Their numbering ranged globally from 100 to 254, but now that the mega repository has passed the 115 plugin ID mark, there is some overlap. Sometimes in the sources you can find a reference to the previous Playground ID, but as there is not much maintenance in the PluginPlayground, once moved to the main repository, the old sources are kind of 'forgotten'. Actually, some maintenance should be done on the PluginPlayground, removing all migrated plugins to avoid confusion.

Reply:
OK, so no issue to be expected when starting a P115 plugin in main repository. Nonetheless i am currently investigating the migration options with P039 in a separate branch. Doesn't look too bad, although the synergies seem lower than expected. At the moment i just see effort - no blocking points. I keep you posted.

Reply:
I think merging this functions to P039 has been successful so far. As i am lacking the possibility to test the thermocouple parts i can only state "think". I enhanced the highlevel SPI driver functions in such a way, they can be re-used in the exiting parts as well and introduced those functions as well to the thermocouple calls. ->  I understood ROM consumption is critical and synergies vital. I will provide those changes in a separate pull request as mentioned earlier. Details to be found there. First tests of merged RTD functions within P039 in test set D have been passed successful on my set up. I ill review my changes on the thermocouple parts with the respecitve data shets and the provide the pull request.

Reply:
Since you're somewhat inexperienced with Git, please make a ZIP first of your code before creating a new branch for a new PR.
The Git commands you might want to know:
- git stash (files may appear to be reverted, but the uncommitted files are placed in a 'virtual' commit)
- git pop (undo of git stash, may also be used to 'move' uncommitted changes to another branch)

Reply:
Um, sorry, i was too fast and not reading this message in time. 

I already started providing a second pull request - hope this works and does not endanger the overall development.
I derived a 2nd branch of mega and migrated/ update the file there.

Reply:
Trust me, you cannot endanger the overall development. :)
You're working on your own fork.
Just tried to warn you about not loosing precious time when still learning Git.
As in... been there done that :)

Reply:
I am fine with every hint supporting my way.

Btw. i tend to close this pull request, as i would prefer to follow up on the functionality as merger to P039. 
What do you think ?

Reply:
Yep, sounds good.